codegen/smithy-aws-go-codegen/src/main/java/software/amazon/smithy/aws/go/codegen/customization/DisableEndpointHostPrefix.java

```diff
@@ -0,0 +1,45 @@
+package software.amazon.smithy.aws.go.codegen.customization;
+
+import java.util.Optional;
+
+
+import software.amazon.smithy.go.codegen.GoSettings;
+import software.amazon.smithy.go.codegen.GoWriter;
+import software.amazon.smithy.go.codegen.SmithyGoDependency;
+import software.amazon.smithy.go.codegen.SymbolUtils;
+import software.amazon.smithy.go.codegen.integration.GoIntegration;
+import software.amazon.smithy.model.Model;
+import software.amazon.smithy.model.pattern.SmithyPattern.Segment;
+import software.amazon.smithy.model.traits.EndpointTrait;
+import software.amazon.smithy.model.shapes.OperationShape;
+
+
+public class DisableEndpointHostPrefix implements GoIntegration {
+    @Override
+    public void renderPostEndpointResolutionHook(
+        GoSettings settings, GoWriter writer, Model model, Optional<OperationShape> operation
+    ) {
+        if (!S3ModelUtils.isServiceS3Control(model, settings.getService(model))) {
+            return;
+        }
+
+        if (operation.isPresent() && operation.get().getTrait(EndpointTrait.class).isPresent()) {
+            EndpointTrait endpointTrait = operation.get().getTrait(EndpointTrait.class).get();
+            boolean written = false;
+            for (Segment segment : endpointTrait.getHostPrefix().getLabels()) {
+                if (segment.isLabel() && segment.getContent().equals("AccountId") && !written) {
+                    writer.write(
+                        """
+                        ctx = $T(ctx, true)
+                        """,
+                        SymbolUtils.createPointableSymbolBuilder(
+                            "DisableEndpointHostPrefix", SmithyGoDependency.SMITHY_HTTP_TRANSPORT).build()
+                    );
+                    // we only want the content above written ONCE per operation.
+                    written = true;
```

Comment:
**nit**: Then just write it and break out of the loop

Reply:
right, will do, in next PR -- artifact left over from my previous logic. a break is more straightforward